EIPS/eip-5089.md

```diff
@@ -0,0 +1,340 @@
+
+Note that most implementations will require pre-approval of the Principal Token contract with the Principal Tokens `underlying` token.
+
+```yaml
+- name: mint
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: amount
+      type: uint256
+    - name: to
+      type: address
+
+  outputs:
+    - name: success
+      type: bool
+```
+
+#### redeem
+
+After maturity, burns exactly `amount` of Principal Tokens from `from` and sends the _same_ `amount` of underlying tokens to `to`.
+_Alberto: It sends underlying tokens according to `exchangeRate`._
+
+OPTIONAL - This method can be used to improve the usability and integration difficulty of Principal Tokens, but interfaces and other contracts MUST NOT expect fund custody to be present.
+While custodial redemption of Principal Tokens through the Principal Token contract is extremely useful for integrators, some protocols may find giving the Principal Token itself custody breaks their backwards compatability. 
+_Alberto: Optionality is evil._
+
+MUST emit the `Redeem` event.
+
+MUST support a redeem flow where the Principal Tokens are burned from `owner` directly where `owner` is `msg.sender` or `msg.sender` has ERC-20 approval over the shares of `owner`.
+MAY support an additional flow in which the shares are transferred to the Principal Token contract before the `redeem` execution, and are accounted for during `redeem`.
+
+MUST revert if all of `amount` cannot be redeemed (due to withdrawal limit being reached, slippage, the owner not having enough Principal Tokens, etc).
+
+Note that some implementations will require pre-requesting to the Principal Token before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: redeem
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: amount
+      type: uint256
+    - name: to
+      type: address
+    - name: from
+      type: address
+
+  outputs:
+    - name: assets
+      type: uint256
+```
+
+#### mature
+_Alberto: We surely don't need this method in the standard. It is up to each PT creator to decide how they mature it._
+
+Callable once `block.timestamp >= maturity`, stores the current `exchangeRate` as the `maturityRate` and "matures" the Principal Tokens. 
+
+OPTIONAL - This method can be used to provide needed data for the calculation of interest post-maturity, but interfaces and other contracts MUST NOT expect this method to be present.
+
+MUST emit the `Mature` event.
+
+MUST revert if `block.timestamp < maturity`.
+
+```yaml
+- name: mature
+  type: function
+  stateMutability: nonpayable
+
+  inputs: []
+
+  outputs:
+    - name: exchangeRate
+      type: uint256
+```
+
+### Events
+
+#### Mint
+
+`from` has exchanged `amount` of underlying for `amount` of Principal Tokens, and transferred those Principal Tokens to `to`.
```

Comment:
If we remove `mint()`, this is unnecessary / would just be covered in the inherited ERCs?

Reply:
Yeah, it would be covered. Removed.